ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/transactional/TransactionalInterceptorBase.java

```diff
@@ -45,17 +46,36 @@
  * @author paul.robinson@redhat.com 02/05/2013
  */
 
-public class TransactionalInterceptorBase implements Serializable {
+public abstract class TransactionalInterceptorBase implements Serializable {
 
     private static final long serialVersionUID = 1L;
 
     @Inject
     transient javax.enterprise.inject.spi.BeanManager beanManager;
 
-    private boolean previousUserTransactionAvailability;
-
     private static TransactionManager transactionManager;
 
+    private final boolean userTransactionAvailable;
+
+    protected TransactionalInterceptorBase(boolean userTransactionAvailable) {
+        this.userTransactionAvailable = userTransactionAvailable;
+    }
+
+    public Object intercept(InvocationContext ic) throws Exception {
+
+        final TransactionManager tm = getTransactionManager();
+        final Transaction tx = tm.getTransaction();
+
+        boolean previousUserTransactionAvailability = setUserTransactionAvailable(false);
```

Comment:
I would have expected you to use the value passed into the constructor (by the subclass) instead of false, as in:

boolean previousUserTransactionAvailability = setUserTransactionAvailable(userTransactionAvailable);